### PR TITLE
feat: 128 bit types

### LINKED ISSUE #6: `I128` and `U128`
Should be very performant and easy both reading and writing just using two `getBigUint64` and bitshifts.

### PR DESCRIPTION
**Pull request by Jabolol**
This PR attempts to close #6 by implementing `I128` and `U128`

### CODE REVIEW

**Review by eliassjogreen — CHANGES_REQUESTED** [review]
Thank you! Looks very well done! I am debating wheter or not `byteAlign` should be 8 or 16. Having 8 as the alignment matches the [behaviour of rust](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1bb1d880aefd7568a862e72dcefbfeb3) but that seems to be [inconsistent with C's definition](https://github.com/rust-lang/rust/issues/54341) due to a bug in llvm. Let's change the `byteAlign` to 16, and if the user wishes to do FFI with Rust they will have to override the `byteAlign` and set that to 8 themselves.

File: types/primitive/i128.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { AlignedType } from "../types.ts";
+import { endianess } from "../../utils.ts";
+
+export class I128 implements AlignedType<bigint> {
+  byteLength = 16;
+  byteAlign = 8;
```
Comment (eliassjogreen) [inline types/primitive/i128.ts]:
```suggestion
  byteAlign = 16;
```

File: types/primitive/u128.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import { AlignedType } from "../types.ts";
+import { endianess } from "../../utils.ts";
+
+export class U128 implements AlignedType<bigint> {
+  byteLength = 16;
+  byteAlign = 8;
```
Comment (eliassjogreen) [inline types/primitive/u128.ts]:
```suggestion
  byteAlign = 16;
```

**Review by eliassjogreen — CHANGES_REQUESTED** [review]

**Comment by Jabolol** [comment]
> Thank you! Looks very well done! I am debating wheter or not `byteAlign` should be 8 or 16. Having 8 as the alignment matches the [behaviour of rust](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1bb1d880aefd7568a862e72dcefbfeb3) but that seems to be [inconsistent with C's definition](https://github.com/rust-lang/rust/issues/54341) due to a bug in llvm. Let's change the `byteAlign` to 16, and if the user wishes to do FFI with Rust they will have to override the `byteAlign` and set that to 8 themselves.

Alright! I'll commit the suggested changes

**Review by eliassjogreen — APPROVED** [review]

**Pull request merged**